design/sketch.webidl

```diff
@@ -89,12 +89,66 @@ enum GPUTextureDimension {
     "3d"
 };
 
+// Texture formats
+// The name of the format specifies the order of components, bits per component, and data type for
+// the component.
+//     r, g, b, a = red, green, blue, alpha
+//     unorm = unsigned normalized
+//     snorm = signed normalized
+//     uint = unsigned int
+//     sint = signed int
+//     float = floating point
+// If the format has the "-srgb" suffix, then sRGB gamma compression and decompression are
+// applied during the reading and writing of color values in the pixel.
+
 enum GPUTextureFormat {
-    "r8g8b8a8-unorm",
-    "r8g8b8a8-uint",
-    "b8g8r8a8-unorm",
-    "d32-float-s8-uint"
-    // TODO other formats
+    /* Normal 8 bit formats */
+    "a8unorm",
+    "r8unorm",
+    "r8unorm-srgb",
+    "r8snorm",
+    "r8uint",
+    "r8sint",
+    /* Normal 16 bit formats */
+    "r16unorm",
+    "r16snorm",
+    "r16uint",
+    "r16sint",
+    "r16float",
+    "rg8unorm",
+    "rg8unorm-srgb",
+    "rg8snorm",
+    "rg8uint",
+    "rg8sint",
+    /* Normal 32 bit formats */
+    "r32uint",
+    "r32sint",
+    "r32float",
+    "rg16unorm",
+    "rg16snorm",
+    "rg16uint",
+    "rg16sint",
+    "rg16float",
+    "rgba8unorm",
+    "rgba8unorm-srgb",
+    "rgba8snorm",
+    "rgba8uint",
+    "rgba8sint",
+    "bgra8unorm",
+    "bgra8unorm-srgb",
+    /* Normal 64 bit formats */
+    "rg32uint",
+    "rg32sint",
+    "rg32float",
+    "rgba16unorm",
+    "rgba16snorm",
+    "rgba16uint",
+    "rgba16sint",
+    "rgba16float",
+    /* Normal 128 bit formats */
+    "rgba32uint",
+    "rgba32sint",
+    "rgba32float",
```

Comment:
Are there no BGRA, ARGB, or ABGR texture types? Are these *actually* RGBA32 (in that they are one 32-bit word which is host-endianness-specific?), or are they R8G8B8A8? Does it matter since POWER is basically dead?